src/Nixfmt/Types.hs

```diff
@@ -184,15 +254,12 @@ class LanguageElement a where
   -- returned. This is useful for getting/extracting values
   mapFirstToken' :: (forall b. Ann b -> (Ann b, c)) -> a -> (a, c)
 
-  -- Map the last token of that expression, no matter how deep it sits
-  -- in the AST. This is useful for modifying comments
-  mapLastToken :: (forall b. Ann b -> Ann b) -> a -> a
-  mapLastToken f a = fst (mapLastToken' (\x -> (f x, ())) a)
-
   -- Same as mapLastToken, but the mapping function also yields a value that may be
   -- returned. This is useful for getting/extracting values
   mapLastToken' :: (forall b. Ann b -> (Ann b, c)) -> a -> (a, c)
 
+  mapAllTokens :: (forall b. Ann b -> Ann b) -> a -> a
+
```

Comment:
It's unfortunate that this needs to be so manual. I've played around with some changes to improve that, very WIP though, just for future reference: https://github.com/NixOS/nixfmt/commit/3b61d8a35b72d87a3dd2ca086f0e4bb13ee5e459